students/CCSt130/lesson03/mail_room_pt2.py

```diff
@@ -0,0 +1,413 @@
+        if amt_input.isdigit():
+            new_amt = float(amt_input)
+            # Error handling should unrealistic amts be entered
+            if new_amt > 0 and new_amt < 1000000000: # n < billion
+                # append to dict
+                all_donations[a_donor][date_str] = new_amt
+                # Replaces all entries for existing donor
+                # all_donations[a_donor] = {dt: new_amt}
+                break
+            else:
+                print()
+                print("Amount must be greater than zero and less than a billion--try again!")
+                print()
+        else: # Non-numeric entry
+            print()
+            print("Invalid entry--try again!") 
+            print()
+    # Inform user
+    print()
+    print(">>> Donation Has Been Entered for {}. <<<".format(a_donor))
+    print()
+    print("###")
+    
+    return(a_donor) # Return user's choice
+
+def list_donor_donations(all_donations, a_donor):
+    """Prints formatted donation details for a specific donor. """
+    # Print header and column labels
+    donation_format = {'Heading1': "Your Donation History:\n",
+                       'Column1': 'Date', 
+                       'Column2': '| Donor', 
+                       'Column3': '| Amount'}
+
+    print()
+    print(donation_format['Heading1'])
+    # Print column labels        
+    print('{:<11} {:<36} {}'.format(donation_format['Column1'], donation_format['Column2'], donation_format['Column3']))
+    print(66 * '-') # Print footer
+    
+    # Print donation dates and amounts (subkeys)
+    for gift_dt, gift_amt in sorted(all_donations[a_donor].items()):
+        print('{:<13} {:<36} ${:>14,.2f}'.format(gift_dt, a_donor, gift_amt))
+            
+    return
+
+def prn_an_email(all_donations):
+    """ Prints formatted email template to screen with donation details. """
+    # Launch function to determine donor
+    a_donor = donor_sub_menu(all_donations)
+    # For today's date to be printed in email
+    from datetime import date
+    # Today's date
+    the_date = str(date.today())
+
+    print('\n\n')
+    print(the_date)
+    print('\n\n')
+    # Template for email
+    donor_email = {'Salutation': "Dear {},\n",
+                   'Body': "We are thrilled to have been the recipient of your generosity! \
+                   \nThank you for your donations to FfFH and your dedication to Feline Health. \
+                   \nYour donations will be put to good use keeping kitties healthy! ",
+                   'Sig_block': "\nBlessings,\n\n\n\nChuck Stevens\nDirector Emeritus\nFoundation for Feline Health"}
+    # Print salutation and body of email
+    print(donor_email['Salutation'].format(a_donor))
+    print(donor_email['Body'])
+   
+    # List all relevant donations in the email generated
+    list_donor_donations(all_donations, a_donor)
+    
+    # Print signature block   
+    print(donor_email['Sig_block'])
+    
+    return
+
+def prn_ltrs(all_donations):
```

Comment:
Ideally, your function names and variables should be self explanatory and explicit.  i.e., print_letters()